techlibs/xilinx/synth_xilinx.cc

```diff
@@ -235,32 +253,33 @@ struct SynthXilinxPass : public Pass
 			Pass::call(design, "dffsr2dff");
 			Pass::call(design, "dff2dffe");
 			Pass::call(design, "opt -full");
-			if (vpr)
-			{
-				Pass::call(design, "techmap -map +/techmap.v -map +/xilinx/arith_map.v -D _EXPLICIT_CARRY");
-				Pass::call(design, "hierarchy -check");
-			}
-			else
-			{
-				Pass::call(design, "techmap -map +/techmap.v -map +/xilinx/arith_map.v");
-			}
+
+            if (vpr) {
+    			Pass::call(design, "techmap -map +/techmap.v -map +/xilinx/arith_map.v -D _EXPLICIT_CARRY");
+            } else {
+    			Pass::call(design, "techmap -map +/techmap.v -map +/xilinx/arith_map.v -map +/xilinx/ff_map.v");
+            }
+
+			Pass::call(design, "hierarchy -check");
 			Pass::call(design, "opt -fast");
 		}
 
 		if (check_label(active, run_from, run_to, "map_luts"))
 		{
-			Pass::call(design, "abc -luts 2:2,3,6:5,10,20" + string(retime ? " -dff" : ""));
```

Comment:
This is wrong, use the same LUT mapping for both flows.

Reply:
Ok. I'll work on that.

Reply:
Fixed.